src/panels/lovelace/editor/lovelace-editor/hui-lovelace-editor.ts

```diff
@@ -0,0 +1,88 @@
+import { html, LitElement, PropertyDeclarations } from "@polymer/lit-element";
+import { TemplateResult } from "lit-html";
+import "@polymer/paper-input/paper-input";
+
+import { EditorTarget } from "../types";
+import { hassLocalizeLitMixin } from "../../../../mixins/lit-localize-mixin";
+import { HomeAssistant } from "../../../../types";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import { configElementStyle } from "../config-elements/config-elements-style";
+
+import { LovelaceConfig } from "../../../../data/lovelace";
+
+declare global {
+  interface HASSDomEvents {
+    "lovelace-config-changed": {
+      config: LovelaceConfig;
+    };
+  }
+}
+
+export class HuiLovelaceEditor extends hassLocalizeLitMixin(LitElement) {
+  static get properties(): PropertyDeclarations {
+    return { hass: {}, _config: {} };
+  }
+
+  get _title(): string {
+    if (!this._config) {
+      return "";
+    }
+    return this._config.title || "";
+  }
+
+  public hass?: HomeAssistant;
+  private _config?: LovelaceConfig;
+
+  set config(config: LovelaceConfig) {
+    this._config = config;
+  }
+
+  protected render(): TemplateResult {
+    if (!this.hass) {
+      return html``;
+    }
+
+    return html`
+      ${configElementStyle}
+      <div class="card-config">
+        <paper-input
+          label="Title"
+          value="${this._title}"
+          .configValue="${"title"}"
+          @value-changed="${this._valueChanged}"
+        ></paper-input>
+      </div>
+    `;
+  }
+
+  private _valueChanged(ev: Event): void {
+    if (!this._config || !this.hass) {
+      return;
+    }
+
+    const target = ev.currentTarget! as EditorTarget;
+
+    if (this[`_${target.configValue}`] === target.value) {
+      return;
+    }
+
+    let newConfig;
+
+    if (target.configValue) {
+      newConfig = {
+        ...this._config,
+        [target.configValue]: target.value,
+      };
+    }
+
+    fireEvent(this, "lovelace-config-changed", { config: newConfig });
```

Comment:
I think that we should make these events not bubble. That way only the parent can listen to it and that would make it faster?

Guess it doesn't really matter in the grand scheme of things.